src/main/java/net/minecraftforge/registries/NewRegistryEvent.java

```diff
@@ -62,8 +63,14 @@ void fill()
                 .collect(Collectors.toMap(RegistryData::builder, d -> d.builder().create(), (a, b) -> a, IdentityHashMap::new));
 
         builtRegistries.forEach((builder, reg) -> {
-            if (builder.getHasWrapper() && !Registry.REGISTRY.containsKey(reg.getRegistryName()))
+            if (builder.isDataPackRegistry())
+            {
+                if (builder.getHasWrapper() && !BuiltinRegistries.REGISTRY.containsKey(reg.getRegistryName()))
+                    RegistryManager.registerToBuiltinRegistry((ForgeRegistry<?>) reg);
+            } else if (builder.getHasWrapper() && !Registry.REGISTRY.containsKey(reg.getRegistryName()))
```

Comment:
Small thing but this would follow the formatting by having the `else if` start on a new line separate from the closing brace.